src/main/java/uk/gov/pay/webhooks/validations/CallbackUrlService.java

```diff
@@ -0,0 +1,59 @@
+package uk.gov.pay.webhooks.validations;
+
+import com.google.common.net.InternetDomainName;
+import org.eclipse.persistence.queries.Call;
+import uk.gov.pay.webhooks.app.WebhooksConfig;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Optional;
+import java.util.Set;
+
+import static java.util.stream.Collectors.toUnmodifiableSet;
+
+public class CallbackUrlService {
+    private final Set<InternetDomainName> allowedDomains;
+
+    public CallbackUrlService(WebhooksConfig webhooksConfig) {
+        this.allowedDomains = webhooksConfig.getLiveDataAllowHosts().stream().map(InternetDomainName::from).collect(toUnmodifiableSet());
+    }
+
+    public void validateCallbackUrl(String callbackUrl, Boolean contextIsLive) throws MalformedURLException {
+        var url = validateAndGetUrlIsWellFormed(callbackUrl);
+        if (Boolean.TRUE.equals(contextIsLive)) {
+            validateUrlIsInLiveDomains(url);
+        }
+    }
+
+    private URL validateAndGetUrlIsWellFormed(String callbackUrl) throws MalformedURLException {
+        URL url;
+        try {
+            url = new URL(callbackUrl);
+        } catch (MalformedURLException e) {
+            throw new CallbackUrlMalformedException("Callback URL is not a valid URL");
+        }
+
+        if (url.getHost().isEmpty()) {
```

Comment:
We’re relying on the not-entirely-clearly-specified behaviour that `getHost()` returns `""` rather than `null` if there is no host.

Reply:
🤔 yeah maybe we just remove this, I'm fairly confident the URL won't be instantiated if there's no host but we could double check that behaviour

Reply:
No, I think we need it. Hostless URLs are a thing. For example, `file:///etc/etc` has no host (but it does have a protocol and a path etc.).